Feat/users groups manager

MamelWorks opened the pull request:
Deploy:
https://github.com/MamelWorks/Laravuecrud/tree/feat/users_groups_manager#deployment

MamelWorks committed:
Addition of Users and Groups manager respecting REST Architectural Constraints
Add: views for user and group management
Add: controls for user and group management
Add: bootstrap-vue-3 and vue-toaster package

Rem: unused files

MamelWorks committed:
fix: make error toast red

tomaszmrozinski commented on database/migrations/2022_11_13_165837_create_group_user_table.php:
```diff
@@ -0,0 +1,33 @@
+    public function up()
+    {
+        Schema::create('group_user', function (Blueprint $table) {
+            $table->id();
```
@MamelWorks group_user map table does not need ID column, user_id and group_id should be primary key ($table->primary). Where are the Foreign keys?

MamelWorks replied:
I agree, both of these columns should be primary key and have foreign key constraints. I'll change that.

tomaszmrozinski commented on app/Http/Controllers/GroupController.php:
```diff
@@ -0,0 +1,45 @@
+    }
+    public function store(Request $request): \Illuminate\Http\JsonResponse
+    {
+        $group = new Group([
```
Where do you validate $request? I don't see any validators.

MamelWorks replied:
So far, there are none, I will add them asap.

tomaszmrozinski commented on app/Http/Controllers/GroupController.php:
```diff
@@ -0,0 +1,45 @@
+        $group->save();
+        return response()->json('Group '. $group->name .' has been created!');
+    }
+    public function show($id)
```
@MamelWorks It's better to use uuids in the frontend not regular IDs - security reasons (this applies to all the code)
Please add proper augments types to all methods, eg.: public function show (string $uuid)

MamelWorks replied:
Should user ID's be replaced with user uuid's? I can see that would create some database performance issues in the future and also readability problems for maintenance team (through communication with client or debugging sessions).

Or should user uuid's be generated and stored beside normal user ID's, where uuid will be used only for communication between backend and frontend?

tomaszmrozinski replied:
@MamelWorks All relations should use regular IDs but Uuids should be used in frontend. Uuid should be indexed in DB:
$table->uuid('uuid')->unique()->index();

MamelWorks replied:
@tomaszmrozinski I see, I will change that.

tomaszmrozinski commented on app/Http/Controllers/UserController.php:
```diff
@@ -0,0 +1,48 @@
+        return response()->json(array_reverse($users));
+    }
+    public function store(Request $request): \Illuminate\Http\JsonResponse
+    {
```
@MamelWorks again no validators - please add validatiors to all update/store methods

MamelWorks replied:
Ok, I'm on it.

tomaszmrozinski commented on app/Http/Controllers/UserController.php:
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Http\Controllers;
```
@MamelWorks uuids instead of IDs

MamelWorks committed:
Add: Back-end and front-end validation of forms.
Change: Uuid's will be now used instead of ID's on front-end for security reasons.
Change: removal of Id and addition of foreign and primary keys in group_user table.

MamelWorks commented:
I'm still thinking about that removal of GroupUserController and UserGroupController controllers.
Changes to it should appear in the evening.
